### PR TITLE
Position + WindowSize

### PR DESCRIPTION
**Pull request by farism**
- Added a `Position` component for positioning an element relative to another target element
- Added a `WindowSize` HOC with a performant `resize` handler. This provides the current window width + height as a `window` prop to the child
- Both of these components are undocumented for now
- Removed documentation for `Portal`
- `Position` now owns the `placements`, and `Popover` and `Tooltip` import placements from `Position`

screen shot 2017-10-27 at 3 26 16 pm
screen shot 2017-10-27 at 3 26 22 pm
screen shot 2017-10-27 at 3 26 31 pm

### CODE REVIEW

File: packages/core-react/src/components/WindowSize/WindowSize.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  throttle(() => subscribers.forEach(sub => sub.resize()), 100)
+)
+
+const getWindowSize = () => ({
+  width: window.innerWidth,
+  height: window.innerHeight,
+})
+
+export default Component => {
+  class WindowSize extends React.Component {
```
Comment (mboperator) [inline packages/core-react/src/components/WindowSize/WindowSize.js]: I love this decorator

File: packages/core-react/src/components/WindowSize/WindowSize.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      this.state = getWindowSize()
+    }
+
+    componentDidMount() {
+      subscribers.push(this)
+    }
+
+    componentWillUnmount() {
+      subscribers = subscribers.filter(sub => sub !== this)
```
Comment (mboperator) [inline packages/core-react/src/components/WindowSize/WindowSize.js]: This is a clean way of getting rid of subscribers, but does this remove the event listener?

File: packages/core-react/src/components/Position/story.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      </div>
+    )
+  }
+}
+
+storiesOf('Position', module).add('default', () => {
+  let examples = []
+
+  for (var i = 0; i < 3; i++) {
+    examples = examples.concat([
```
Comment (mboperator) [inline packages/core-react/src/components/Position/story.js]: Didn't realize you could do this. Nice.

File: packages/core-react/src/components/WindowSize/WindowSize.js (hunk 2 above)
Comment (farism) [reply packages/core-react/src/components/WindowSize/WindowSize.js]: It doesn't, the event listener is module scoped :/ so if this is somehow included in two different bundles we would get multiple event listeners. Definitely an improvement over an individual listener for each instance though.

**Pull request merged**